doc_src/language.rst

```diff
@@ -875,14 +875,15 @@ There are three kinds of variables in fish: universal, global and local variable
 - Universal variables are shared between all fish sessions a user is running on one computer.
 - Global variables are specific to the current fish session, and will never be erased unless explicitly requested by using ``set -e``.
 - Local variables are specific to the current fish session, and associated with a specific block of commands, and automatically erased when a specific block goes out of scope. A block of commands is a series of commands that begins with one of the commands ``for``, ``while`` , ``if``, ``function``, ``begin`` or ``switch``, and ends with the command ``end``.
+- Function-local variables are the same as local variables outside of one of these blocks. They go out of scope when the current function ends.
```

Comment:
I suggest moving this up above `--local`, since now we want to de-emphasize local variables. Also we should not use "function-local" since that risks some confusion with local variables. I think "function variables" is fine.

My attempt:

> Function variables are scoped to the body of the currently executing function, and are erased when the function completes.
>
> Local variables are scoped to the body of the current innermost `for`, `while`, `if`, `begin`, `switch`, or `function` command, and are erased when that command completes. If the command is `function`, then it is a function variable.